src/SnapKeyring.ts

```diff
@@ -457,120 +577,65 @@ export class SnapKeyring extends EventEmitter {
-  /**
-   * Get all accounts associated with a snap ID.
-   *
-   * @param snapId - The snap ID to get accounts for.
-   * @returns All accounts associated with the given snap ID.
-   */
-  #getAccountsBySnapId(snapId: string): KeyringAccount[] {
-    return [...this.#addressToAccount.values()].filter(
-      (account) => this.#addressToSnapId.get(account.address) === snapId,
+    return (
+      Array.from(this.#accounts.values()).find(({ account }) =>
+        equalsIgnoreCase(account.address, address),
+      ) ?? throwError(`Account '${address}' not found`)
     );
   }
 
   /**
-   * Add an account to the internal maps.
-   *
-   * @param snapAccount - The account to be added.
-   * @param snapId - The snap ID of the account.
-   */
-  #addAccountToMaps(snapAccount: KeyringAccount, snapId: string): void {
-    const account = {
-      ...snapAccount,
-      // FIXME: Do not lowercase the address here. This is a workaround to
-      // support the current UI which expects the account address to be
-      // lowercase. This workaround should be removed once we migrated the UI
-      // to use the account ID instead of the account address.
-      address: snapAccount.address.toLowerCase(),
-    };
-    this.#addressToAccount.set(account.address, account);
-    this.#addressToSnapId.set(account.address, snapId);
-  }
-
-  /**
-   * Remove an account from the internal maps.
-   *
-   * @param account - The account to be removed.
-   */
-  #removeAccountFromMaps(account: KeyringAccount): void {
-    this.#addressToAccount.delete(account.address);
-    this.#addressToSnapId.delete(account.address);
-  }
-
-  /**
-   * Get an account by its ID. Returns undefined if the account is not found.
+   * Get the metadata of a snap keyring account.
    *
-   * @param id - Account ID.
-   * @returns The account object or undefined if the account is not found.
+   * @param snapId - Snap ID.
+   * @returns The snap metadata or undefined if the snap cannot be found.
    */
-  #getAccountById(id: string): KeyringAccount | undefined {
-    return [...this.#addressToAccount.values()].filter(
-      (account) => account.id === id,
-    )[0];
-  }
-
   #getSnapMetadata(
-    address: string,
+    snapId: string,
   ): InternalAccount['metadata']['snap'] | undefined {
-    const snapId = this.#addressToSnapId.getOrThrow(address);
     const snap = this.#snapClient.getController().get(snapId);
     return snap
       ? { id: snapId, name: snap.manifest.proposedName, enabled: snap.enabled }
       : undefined;
   }
 
+  /**
+   * Remove all accounts associated with a given Snap ID.
+   *
+   * @param snapId - The Snap ID to remove accounts for.
+   * @returns A Promise that resolves when all accounts have been removed.
+   */
+  async removeAccountsBySnapId(snapId: string): Promise<void> {
+    for (const entry of this.#accounts.values()) {
+      if (entry.snapId === snapId) {
+        await this.removeAccount(entry.account.address.toLowerCase());
+      }
+    }
+  }
```

Comment:
Do you know if `snapIds` are case sensitive?

Reply:
the ids should be npm package names with the `npm:` prefix, and the package names should always be lowercase.